### PR TITLE
Add script to flash WinPE for i.MX8M.

### PR DESCRIPTION
**Pull request by liviu-razorea**
Add script to flash WinPE for i.MX8M.

### CODE REVIEW

File: build/tools/make-winpe-i.MX8.cmd [hunk 1]
```diff
@@ -0,0 +1,234 @@
+::
+:: Prepares a WinPE image for boot on i.MX6 or i.MX7
```
Comment (neilsh-msft) [inline build/tools/make-winpe-i.MX8.cmd]: update comment to indicate this builds for iMX8

File: build/tools/make-winpe-i.MX8.cmd [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+popd
+
+:: make directory to hold packages
+rmdir /s /q packages > NUL 2>&1
+mkdir packages
+pushd packages
+set PACKAGES_DIR=%CD%
+popd
+
+:: Parse options
```
Comment (neilsh-msft) [inline build/tools/make-winpe-i.MX8.cmd:13]: The options should be parsed before any work is done.  Before this point the script deletes the mount and packages directories in the local directory.  If I run make-winpe-i.mx8.cmd /?  it should just print out the help text, and not do anything else.

File: build/tools/make-winpe-i.MX8.cmd [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+    echo Creates a WinPE image for i.MX
+    echo Options:
+    echo.
+    echo    /builddir build_dir          Path to build output directory.
+    echo    /firmware firmware_fit_path  Path to firmware_fit.merged
+    echo    /uefi uefi_fit_path          Path to uefi.fit
+    echo    /ffu ffu_path                Optionally specify an FFU to flash
+    echo    /ffudisk ffu_disk_number     Optionally specify the physical disk
+    echo                                 number to which the FFU is applied.
+    echo                                 Defaults to 1.
```
Comment (neilsh-msft) [inline build/tools/make-winpe-i.MX8.cmd]: You changed the default to 0, update the message here.

File: build/tools/make-winpe-i.MX8.cmd [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+    echo make-winpe.cmd /builddir build_dir /firmware firmware_fit_path
+    echo   /uefi uefi_fit_path [/ffu ffu_path] [/ffudisk ffu_disk_number]
+    echo make-winpe.cmd /apply disk_number
+    echo make-winpe.cmd /clean
+    echo.
+    echo Creates a WinPE image for i.MX
+    echo Options:
+    echo.
+    echo    /builddir build_dir          Path to build output directory.
+    echo    /firmware firmware_fit_path  Path to firmware_fit.merged
```
Comment (neilsh-msft) [inline build/tools/make-winpe-i.MX8.cmd]: If firmware_fit.merged the filename used for the iMX8?  I think it's flash.bin.

File: build/tools/make-winpe-i.MX8.cmd [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+
+echo Copying driver packages to %PACKAGES_DIR%
+call :GETPACKAGE NXP.iMX.HalExtiMXDma.cab || goto err
+call :GETPACKAGE NXP.iMX.imxecspi.cab || goto err
+call :GETPACKAGE NXP.iMX.imxgpio.cab || goto err
+call :GETPACKAGE NXP.iMX.imxi2c.cab || goto err
+call :GETPACKAGE NXP.iMX.imxnetmini.cab || goto err
+call :GETPACKAGE NXP.iMX.imxuart.cab || goto err
+call :GETPACKAGE NXP.iMX.imxusdhc.cab || goto err
+
```
Comment (neilsh-msft) [inline build/tools/make-winpe-i.MX8.cmd:128]: For what reason did you remove the NXP.iMX.Optee.Tree cab for the iMX8?

**Review by neilsh-msft — CHANGES_REQUESTED** [review]

File: build/tools/make-winpe-i.MX8.cmd [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+)
+xcopy /herky "%WINPE_DIR%\Media" "%DEST%\" || goto err
+copy "%WINPE_DIR%\en-us\winpe.wim" "%DEST%\sources\boot.wim" || goto err
+move "%DEST%\bootmgr.efi" "%DEST%\EFI\Microsoft\boot\" || goto err
+
+:: BCD
+echo Enabling kernel debugging
+set TARGET_BCD_STORE=%DEST%\EFI\Microsoft\boot\bcd
+bcdedit /store "%TARGET_BCD_STORE%" /dbgsettings SERIAL DEBUGPORT:1 BAUDRATE:115200 || goto err
+bcdedit /store "%TARGET_BCD_STORE%" /debug {default} on || goto err
```
Comment (christopherco) [inline build/tools/make-winpe-i.MX8.cmd]: Do we want to have KD enabled in the WinPE image by default?

Comment (liviu-razorea) [reply build/tools/make-winpe-i.MX8.cmd]: I added a new command-line argument to enable it. It's not on by default anymore.

File: build/tools/make-winpe-i.MX8.cmd (hunk 5 above)
Comment (liviu-razorea) [reply build/tools/make-winpe-i.MX8.cmd:128]: Enabling NXP.iMX.Optee.Tree is not part of the goal of this PR.

**Review by neilsh-msft — APPROVED** [review]

**Pull request merged**